Sources/WhisperKit/Core/WhisperKit.swift

```diff
@@ -211,114 +228,80 @@ open class WhisperKit {
         }
     }
 
-    /// Sets up the model folder either from a local path or by downloading from a repository.
-    public func setupModels(
-        model: String?,
-        downloadBase: URL? = nil,
-        modelRepo: String?,
-        modelFolder: String?,
-        download: Bool
-    ) async throws {
-        // Determine the model variant to use
-        let modelVariant = model ?? WhisperKit.recommendedModels().default
-
-        // If a local model folder is provided, use it; otherwise, download the model
-        if let folder = modelFolder {
-            self.modelFolder = URL(fileURLWithPath: folder)
-        } else if download {
-            let repo = modelRepo ?? "argmaxinc/whisperkit-coreml"
-            do {
-                self.modelFolder = try await Self.download(
-                    variant: modelVariant,
-                    downloadBase: downloadBase,
-                    useBackgroundSession: useBackgroundDownloadSession,
-                    from: repo
-                )
-            } catch {
-                // Handle errors related to model downloading
-                throw WhisperError.modelsUnavailable("""
-                Model not found. Please check the model or repo name and try again.
-                Error: \(error)
-                """)
-            }
-        }
-    }
-
     public func prewarmModels() async throws {
         try await loadModels(prewarmMode: true)
     }
 
     public func loadModels(
         prewarmMode: Bool = false
     ) async throws {
-        modelState = prewarmMode ? .prewarming : .loading
+        assert(modelFolder != nil || mlxModelFolder != nil, "Please specify `modelFolder` or `mlxModelFolder`")
```

Comment:
Will this crash on prod builds? If so it should throw instead of crash.

Reply:
it's not going to crash on a prod build